nicknamer.py

```diff
@@ -175,4 +180,64 @@ async def trace(context: Context) -> None:
         )
 
 
+@nicknamer.event
+async def on_message(message: Message) -> None:
+    cleaned_urls = {}
+
+    take_extreme_counter_measures = False
+
+    extractor = URLExtract()
+    for url in extractor.gen_urls(message.content):
+        clean_url = clear_url(url)
+
+        if url != clean_url:
+            cleaned_urls[url] = clean_url
+
+        take_extreme_counter_measures = take_extreme_counter_measures or len(
+            clean_url) * URL_LENGTH_VIOLATION_FACTOR < len(url)
+
+    if cleaned_urls:
+        jar_jar_emoji = await message.channel.guild.fetch_emoji(JAR_JAR_EMOJI_ID)
+        await message.add_reaction(jar_jar_emoji)
+
+        if take_extreme_counter_measures and message.author.id == ZACH_USER_ID:
+            await asyncio.sleep(.2)
+
+            jar_jar_embed = Embed(title="Jar Jar Link Countermeasures",
+                                  description="Icky icky linky", color=JAR_JAR_COLOR_HEX)
+            jar_jar_embed.set_thumbnail(url="https://cdn.mos.cms.futurecdn.net/RvLDChLaR37NWTEjvQm2pB-970-80.jpg.webp")
```

Comment:
I don't like this. This file can get moved. We should set up our own hosting as a next step.

Reply:
Agreed.